sources/core/Stride.Core.Reflection/TypeDescriptors/SetDescriptor.cs

```diff
@@ -11,40 +12,50 @@ namespace Stride.Core.Reflection
 {
     public class SetDescriptor : CollectionDescriptor
     {
-        private static readonly List<string> ListOfMembersToRemove = new List<string> { "Comparer", "Capacity" };
-
-        private readonly MethodInfo addMethod;
-        private readonly MethodInfo removeMethod;
-        private readonly MethodInfo clearMethod;
-        private readonly MethodInfo containsMethod;
-        private readonly MethodInfo countMethod;
-        private readonly MethodInfo isReadOnlyMethod;
-
+        private static readonly List<string> ListOfMembersToRemove = ["Comparer", "Capacity"];
+
+        private Action<object, object?> AddMethod;
+        private Action<object, object?> RemoveMethod;
+        private Action<object> ClearMethod;
+        private Func<object, object?,bool> ContainsMethod;
+        private Func<object, int> CountMethod;
+        private Func<object, bool> IsReadOnlyMethod;
+
+        #pragma warning disable CS8618
+        // This warning is disabled because the necessary initialization will occur 
+        // in the CreateSetDelegates<T>() method, not in the constructor.
         public SetDescriptor(ITypeDescriptorFactory factory, Type type, bool emitDefaultValues, IMemberNamingConvention namingConvention)
             : base(factory, type, emitDefaultValues, namingConvention)
         {
             if (!IsSet(type))
-                throw new ArgumentException(@"Expecting a type inheriting from System.Collections.ISet", nameof(type));
-
-            // extract Key, Value types from ISet<??>
-            var interfaceType = type.GetInterface(typeof(ISet<>));
-
-            // Gets the element type
-            ElementType = interfaceType.GetGenericArguments()[0] ?? typeof(object);
-
-            Type[] ArgTypes = { ElementType };
-            addMethod = interfaceType.GetMethod("Add", ArgTypes);
-            removeMethod = type.GetMethod("Remove", ArgTypes);
-            clearMethod = interfaceType.GetMethod("Clear");
-            containsMethod = type.GetMethod("Contains", ArgTypes);
-            countMethod = type.GetProperty("Count").GetGetMethod();
-            isReadOnlyMethod = type.GetInterface(typeof(ICollection<>)).GetProperty("IsReadOnly").GetGetMethod();
+                throw new ArgumentException(@"Expecting a type inheriting from System.Collections.ISet<T>", nameof(type));
```

Comment:
this was just copy paste from the other descriptors like dictionary and list, it seems to be added after the others and implemented the same way even though it only works for generic one as the non generic one doesnt exist or isnt handled